Assets/Scripts/Tank/TankService.cs

```diff
@@ -1,37 +1,97 @@
+        }
+    }
+
     private void Start() {
-        GetJoystick();
-        CreateTank();
-        AssignCamera();
+        populateSpawnPositions();
+        CreatePlayer();
+        EnemyController createdEnemy;
+        createdEnemy = CreateEnemy(enemiesToGenerate[0]);
+        enemyList.Add(createdEnemy);
+        createdEnemy = CreateEnemy(enemiesToGenerate[1]);
+        enemyList.Add(createdEnemy);
+        createdEnemy = CreateEnemy(enemiesToGenerate[2]);
+        enemyList.Add(createdEnemy);
+    }
+
+    // Player Creation...
+    private void CreatePlayer(){
+        Vector3 randomSpawnPosition = GetRandomSpawnPos();
+        playerTank.transform.position = randomSpawnPosition;
+        Joystick joystick = GetJoystick();
+        AssignJoystick(playerTank, joystick);
+        AssignCamera(playerTank);
     }
 
-    private void GetJoystick(){
+    private Joystick GetJoystick(){
         InputService joystickInstance = InputService.Instance;
-        stick = joystickInstance.Joystick;
+        Joystick tempJoystick;
+        tempJoystick = joystickInstance.Joystick;
+        return tempJoystick;
     }
 
-    #region Tank Creator
-        private void CreateTank(){
-            tankController = CreateTankController(stick);
-        }
-        private TankController CreateTankController(Joystick stick){
-            tankModel = new TankModel(10f, 30f, 100);
-            tankController = new TankController(tankModel, tankView, stick);
-            return tankController;
-        }
-    #endregion
+    private void AssignJoystick(TankController tankController,  Joystick joystick){
+        tankController.Joystick = joystick;
+    }
 
-    private void AssignCamera(){
+    private void AssignCamera(TankController player){
         mainCamera = Camera.main;
         GameObject playerObj;
-        playerObj = tankController.TankView.gameObject;
+        playerObj = player.gameObject;
         mainCamera.gameObject.GetComponent<PlayerTracker>().PlayerTransform = playerObj.GetComponent<Transform>();
     }
+
+    // Enemy Creation... Enemy tank factory pattern and controlled by EnemyService
+
+    private void populateSpawnPositions(){
+        spawnPositions.AddRange(spawnerObject.gameObject.GetComponentsInChildren<Transform>());
+    }
+
+    private EnemyController CreateEnemy(TankScriptableObject tankData){
+        TankType type = tankData.type;
+        EnemyController enemy;
+        Vector3 randomSpawnPosition = GetRandomSpawnPos();
+        if(type == TankType.Blue){
+            enemy = GameObject.Instantiate(blueTank, randomSpawnPosition, Quaternion.identity, enemyParent);
+        } else if(type == TankType.Red){
+            enemy = GameObject.Instantiate(redTank, randomSpawnPosition, Quaternion.identity, enemyParent);
+        } else {
+            enemy = GameObject.Instantiate(greenTank, randomSpawnPosition, Quaternion.identity, enemyParent);
+        }
+        enemy.Health = tankData.health;
```

Comment:
Pass scriptable object to enemy controller and setup enemy in that script. Single Responsibility Principle is broken as this function does too many things.

Reply:
will do.